packages/core/src/modules/connections/handlers/ConnectionRequestHandler.ts

```diff
@@ -62,30 +63,31 @@ export class ConnectionRequestHandler implements MessageHandler {
       )
     }
 
-    const receivedDidRecord = await this.didRepository.findReceivedDidByRecipientKey(
-      messageContext.agentContext,
-      senderKey
-    )
+    const receivedDidRecord = await this.didRepository.findReceivedDidByRecipientKey(agentContext, senderKey)
     if (receivedDidRecord) {
       throw new AriesFrameworkError(`A received did record for sender key ${senderKey.fingerprint} already exists.`)
     }
 
     const connectionRecord = await this.connectionService.processRequest(messageContext, outOfBandRecord)
 
+    // Associate the new connection with the session created for the inbound message
```

Comment:
Ideally we don't do transport stuff in the connection handler, but it seems this is an exception and makes sense